esp-hal-common/src/uart.rs

```diff
@@ -1280,88 +1331,94 @@ mod asynch {
     const INIT: AtomicWaker = AtomicWaker::new();
     static WAKERS: [AtomicWaker; NUM_UART] = [INIT; NUM_UART];
 
-    pub(crate) enum Event {
-        TxDone,
-        TxFiFoEmpty,
-        RxFifoFull,
-        RxCmdCharDetected,
-        RxFifoOvf,
+    bitflags! {
+        /// Bitmasks for the INT_ENA, INT_RAW and INT_CLR registers
+        /// Unfortunately, we cannot get them from the svd2rust generated PAC-code, without having to issue a register read first.
+        /// TODO: bit 19 for certain chips
+        /// TODO: Validate bits 0-18 are really the same for all chips
```

Comment:
Part of the reason to use pacs is that for the most part, they abstract over the individual bits, meaning something like `tx_done_int_ena().set_bit()` does the same thing on all chips, even though the bits may be at different places within the register.

Therefore I don't think this `bitflags` is not the right approach here.